builder/vsphere/common/step_add_flag.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type FlagConfig
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-sdk/template/interpolate"
+	"github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/driver"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type FlagConfig struct {
```

Comment:
Curious on the name for this structure. Why FlagConfig? 

It doesn't give any indication that it is related to VBS. I would expect something like VBSSettingsConfig with a StepAddVBSSettings

Reply:
This is for virtual machine Flags - `types.VirtualMachineFlagInfo()` - two of which are VBS (`VBSEnabled`) and IO/MMU (`VvtdEnabled`) flag among others. Then we can allow additional flags to be added if there are additional needs - like VM logging is a flag.

Think of it it like how `HardwareConfig` is mapped to `types.VirtualMachineConfigSpec`.

Reply:
Thanks for the additional context. This was helpful to me.